cmd/karavictl/cmd/generate_admin_token.go

```diff
@@ -0,0 +1,112 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"karavi-authorization/internal/token"
+	"karavi-authorization/internal/token/jwx"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	refreshTokenTTL int64
+	accessTokenTTL  int64
+	password        string
+)
+
+// NewAdminTokenCmd creates a new token command
+func NewAdminTokenCmd() *cobra.Command {
+	adminTokenCmd := &cobra.Command{
+		Use:   "token",
+		Short: "Generate tokens for a admin.",
+		Long:  `Generates tokens for a admin.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			adminName, err := cmd.Flags().GetString("name")
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return err
+			}
+			if strings.TrimSpace(adminName) == "" {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), errors.New("empty admin name not allowed"))
+			}
+
+			refExpTime, err := cmd.Flags().GetDuration("refresh-token-expiration")
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return err
+			}
+			if refExpTime <= 0 {
+				refreshTokenTTL = int64(24 * time.Hour)
+			}
+
+			accExpTime, err := cmd.Flags().GetDuration("access-token-expiration")
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return err
+			}
+			if accExpTime <= 0 {
+				accessTokenTTL = int64(30 * time.Minute)
+			}
+
+			password, err := cmd.Flags().GetString("password")
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return err
+			}
+
+			// If the password was not provided...
+			prompt := fmt.Sprintf("Enter password: ")
+			// If the password was not provided...
+			if pf := cmd.Flags().Lookup("password"); !pf.Changed {
+				// Get password from stdin
+				readPassword(cmd.ErrOrStderr(), prompt, &password)
+			}
+
+			tm := jwx.NewTokenManager(jwx.HS256)
+			// Generate the token.
+			s, err := token.CreateAdminSecret(tm, token.Config{
+				AdminName:         adminName,
+				Subject:           "admin",
+				Roles:             nil,
+				JWTSigningSecret:  password,
+				RefreshExpiration: time.Duration(refreshTokenTTL),
+				AccessExpiration:  time.Duration(accessTokenTTL),
+			})
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return nil
+			}
+
+			err = JSONOutput(cmd.OutOrStdout(), &s)
+			if err != nil {
+				reportErrorAndExit(JSONOutput, cmd.ErrOrStderr(), err)
+				return nil
+			}
+			return nil
+		},
+	}
+
+	adminTokenCmd.Flags().StringP("name", "n", "", "Admin name")
+	adminTokenCmd.Flags().StringP("password", "p", "", "Specify password, or omit to use stdin")
```

Comment:
Instead of using "password", this can be "jwtsigningsecret" (or "jwt-signing-secret") or similar to remain consistent with our RPM and Helm installs.

Reply:
Fixed